x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/optimizer/Optimizer.java

```diff
@@ -983,6 +984,37 @@ public LogicalPlan apply(LogicalPlan p) {
         }
     }
 
+    static class ReplaceSumWithStats extends OptimizerBasicRule {
+        // this is a workaround for the https://github.com/elastic/elasticsearch/issues/45251 issue
+        // should be removed as soon as the issue is fixed
+        // NOTE: this rule should always be applied AFTER the ReplaceAggsWithStats rule
+
+        @Override public LogicalPlan apply(LogicalPlan plan) {
+            final Map<Expression, Stats> statsPerField = new LinkedHashMap<>();
+            
+            plan.forEachExpressionsUp(e -> {
+                if (e instanceof Sum) {
+                    statsPerField.computeIfAbsent(((Sum) e).field(), field -> {
+                        Source source = new Source(field.sourceLocation(), "STATS(" + field.sourceText() + ")");
+                        return new Stats(source, field);
+                    });
+                }
+            });
+            
+            if (statsPerField.isEmpty()) {
```

Comment:
Minor adjustment - to avoid having two return staments do the transformation if the list is non-empty and assign that to the plan

```
if (statsPerField.isEmpty() == false) {
   plan = plan.transformExpression()
}

return plan;
```